xla/service/gpu/gpu_compiler.cc

```diff
@@ -1416,7 +1416,7 @@ absl::Status GpuCompiler::OptimizeHloPostLayoutAssignment(
                          LayoutAssignment::InstructionCanChangeLayout)
                      .VerifyBroadcastDimensionsOrder()
                      .VerifyReshapeIsBitcast(),
-                 /*debug_only=*/true);
+                 /*debug_only=*/false);
```

Comment:
I think this add the HloVerifier after each optimization pass.
Did you look at the history to see if this was explicitly added to speed up the compilation?
If so, may we should add only one pass late that is for non-debug?

Reply:
Indeed, the reason why it was turned off was due to compile time. The idea was that we would still run the verifier for tests, anyway. Unfortunately our tests don't cover everything. So I agree, adding the verifier just for just one pass is probably better.

Reply:
I have added the verifier to the end of this pipeline. Does that look reasonable?

(I have also verified locally that this would catch the PAX crasher that we had.)